### PR TITLE
MTDSA-14925: Added logic for R8B changes

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v3/controllers/RetrieveCalculationController.scala [hunk 1]
```diff
@@ -55,12 +57,24 @@ class RetrieveCalculationController @Inject() (val authService: EnrolmentsAuthSe
...
           .withService(service.retrieveCalculation)
           .withHateoasResultFrom(hateoasFactory) { (request, response) =>
-            RetrieveCalculationHateoasData(
-              nino = nino,
-              taxYear = request.taxYear,
-              calculationId = calculationId,
-              response = response
-            )
+            if(!FeatureSwitches(appConfig.featureSwitches).isR8bSpecificApiEnabled &&
+              response.calculation.exists(calc => calc.endOfYearEstimate.exists(eoy => eoy.totalAllowancesAndDeductions.isDefined))) {
```
Comment (tom-lloyd-hmrc) [inline app/v3/controllers/RetrieveCalculationController.scala:61]: I think Chandra's doing the main review, but just noticed here we could probably simplify this code a little by calculating the response first, and then creating the hateoas data once afterwards.

Comment (narendravijayarao) [reply app/v3/controllers/RetrieveCalculationController.scala:61]: Doesn't make any difference, we will have an extra **_val response_**. Also, I don't want to disturb the existing code much. This release is in April. Just in 1 month we may remove this feature switch completely

**Review by sudoku007 — CHANGES_REQUESTED** [review]

File: app/v3/controllers/RetrieveCalculationController.scala (hunk 1 above)
Comment (sudoku007) [reply app/v3/controllers/RetrieveCalculationController.scala:61]: Yeah I agree we could simplify it a bit by adding a wrapper method to adapt the response based on the feature switch flag,  we could then unit test that method independently too.

File: test/v3/controllers/RetrieveCalculationControllerSpec.scala [hunk 2]
```diff
@@ -79,6 +83,43 @@ class RetrieveCalculationControllerSpec
...
 
+        MockAppConfig.featureSwitches.returns(Configuration("r8b-api.enabled" -> false))
+
+        MockHateoasFactory
+          .wrap(response, RetrieveCalculationHateoasData(nino, TaxYear.fromMtd(taxYear), calculationId, emptyCalculationResponse))
+          .returns(HateoasWrapper(emptyCalculationResponse, hateoaslinks))
+
+        runOkTest(OK, Some(r8bMtdResponseJson))
+      }
+
```
[the remark this answers is not in the document]
Comment (narendravijayarao) [reply test/v3/controllers/RetrieveCalculationControllerSpec.scala:94]: The first test **"happy path with r8b-api disabled"** is checking this scenario

[the remark this answers is not in the document]
Comment (sudoku007) [reply test/v3/controllers/RetrieveCalculationControllerSpec.scala:94]: yeah, missed seeing this

**Review by sudoku007 — APPROVED** [review]

**Pull request merged**